### ISSUE TITLE
When deploying jar to remote repository, file not found.

### ISSUE THREAD

**Issue by stephenpfaff** [mjiderhamn/promote-maven-plugin#2]
This seems to be an issue when deploying jar artifacts to a remote repository.  Using maven-deploy-plugin (deploy:deploy), this does not provide any context during mvn lifecycle in order to find artifacts.  This results in an error...

`The packaging for this project did not assign a file to the build artifact.`

Running just 'deploy' works, but this executes the entire lifecycle which would defeat the purpose of a build promotion.  Need a way to provide back to the process the location for last build artifacts in the workspace or local file system.

**Comment by mjiderhamn** [comment]
Hi Stephen. The `promote:artifacts` goal is supposed to handle this for you. Have you tried `promote:artifacts deploy:deploy`?

If that does not help, what does your config look like?

**Comment by stephenpfaff** [comment]
Thanks for the help and the plugin, I was missing this goal.

**Comment by jeffreyang** [comment]
I was getting a similar error, and "No file registered for artifact: com.test:my-project:jar:1.1-SNAPSHOT".  

It was adding and committing a new release tag to my source code repository, but not deploying the release version to Maven Nexus repository.

It took me a few hours to make it work.  To save others the same pain, here's what I did:
In my pom.xml, note maven-release-plugin's completionGoals:

```
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-release-plugin</artifactId>
    <version>2.5.3</version>
    <configuration>
        <completionGoals>promote:artifacts jar:jar deploy:deploy</completionGoals>
    </configuration>
</plugin>

```

Then I executed the following steps:
1. mvn clean install
2. mvn promote:make-promotable
3. mvn release:clean promote:prepare release:prepare

EDIT:  Actually, this will only work if packaging is "jar".  I cannot make it work for "ear"
